### ISSUE TITLE
CWD set error

### ISSUE THREAD

**Issue by TheLizzard** [TheLizzard/CPP-IDLE#4]
```python
Traceback (most recent call last):
  File "/usr/lib/python3.10/tkinter/__init__.py", line 1921, in __call__
    return self.func(*args)
  File "/path/src/plugins/virtualevents.py", line 97, in <lambda>
    drop_func = lambda e: self.send(event_name, drop=False)
  File "/path/src/plugins/virtualevents.py", line 60, in send
    ret:str = self._send_rest(event_name, drop=False, **kwargs)
  File "/path/src/plugins/virtualevents.py", line 147, in _send_rest
    ret:str = vir_event.send(event_name, other=True, **kwargs)
  File "/path/src/plugins/virtualevents.py", line 79, in send
    result:str = func(event)
  File "/path/src/plugins/rules/baserule.py", line 37, in <lambda>
    func = lambda event, on=better_on: self(event, on)
  File "/path/src/plugins/rules/baserule.py", line 69, in __call__
    data = self.applies(event, on)
  File "/path/src/plugins/rules/runmanager.py", line 45, in applies
    data:str = event.data[0]
IndexError: tuple index out of range
```
The event `"<<Explorer-Set-CWD>>"` was sent to `tkinter` (which dropped the `event.data`) and `VirtualEvents` picked it back up before handling `RunManager`. `RunManager` expected `event.data` to not be empty.
This happened after #3. And the error was duplicated multiple times (maybe 1 for each `RunManager` alive?).

**Comment by TheLizzard** [comment]
In
```python
def _send_rest(self, event_name:str, **kwargs:dict) -> str:
    ret:str = ""
    for vir_event in _vir_events.values():
        if vir_event == self:
            continue
        ret:str = vir_event.send(event_name, other=True, **kwargs)
        if ret == "break":
            return "break"
    return ret
```
If the last `VirtualEvents` object doesn't return `"handled"` or `"break"`, the event get sent to `tkinter` by this:
```python
def send(self, event_name:str, *, drop:bool=True, other:bool=False, **kwargs:dict) -> str:
    if (not other) and drop:
        ret:str = self._send_rest(event_name, drop=False, **kwargs)
        if ret in ("break", "handled"):
            return ret
    ...
    if drop:
        return self.old_event_generate(event_name, **kwargs)
```

Detached plugins (like when a plugin is being destroyed because the tab was closed), still have `VirtualEvents`' objects alive which try to handle `bind_all` events but they can't so they don't return `"handled"` or `"break"`. Which is why the event gets dropped to `tkinter` which strips away the `event.data`. Will be fixed in the next commit.

**Comment by TheLizzard** [comment]
Fixed in Version 2.7
